platform/darwin/src/MGLOfflineStorage.mm

```diff
@@ -74,7 +78,8 @@ - (void)unpauseFileSource:(__unused NSNotification *)notification {
 - (void)setDelegate:(id<MGLOfflineStorageDelegate>)newValue {
     _delegate = newValue;
     if ([self.delegate respondsToSelector:@selector(offlineStorage:URLForResourceOfKind:withURL:)]) {
-        _mbglFileSource->setResourceTransform([offlineStorage = self](auto kind_, std::string&& url_) -> std::string {
+        delete _mbglResourceTransform;
+        _mbglResourceTransform = new mbgl::Actor<mbgl::ResourceTransform>(*mbgl::util::RunLoop::Get(), [offlineStorage = self](auto kind_, const std::string&& url_) -> std::string {
```

Comment:
I’m not sure, but maybe we can replace the `mbglResourceTransform` property with an instance variable [here](https://github.com/mapbox/mapbox-gl-native/blob/46e5b4a0e4e1f4be7958417d9dc87d3763e71fc4/platform/darwin/src/MGLOfflineStorage.mm#L43):

```objc
@implementation MGLOfflineStorage {
    std::unique_ptr<Actor<ResourceTransform>> _mbglResourceTransform;
}
```